tests/test_cats.py

```diff
@@ -17,10 +17,10 @@ class CatTest:
 
 def test_cats_decorator() -> None:
 
-    with pytest.raises(TypeError):
+    with pytest.raises((TypeError, KeyError)):
```

Comment:
`KeyError` gets raised, no TypeError occurs. I will investigate what prompted this change and report back.

Reply:
This error is occurring within the code generated by GenConverter.

Here is the generated code for `CatTest`:
```

def structure_CatTest(o, _, *, __cl=__cl, __c_structure_name=__c_structure_name, __c_structure_age=__c_structure_age, __c_structure_neutered=__c_structure_neutered, __c_structure_alive=__c_structure_alive):  
    res = {} 
    if 'alive' in o:   
        res['alive'] = __c_structure_alive(o['alive']) 
    return __cl(    __c_structure_name(o['name']),    __c_structure_age(o['age']),    __c_structure_neutered(o['neutered']),    **res,  )
```

There is not much we can do to restore back to `ValueError` besides trying to get a change upstream in cattrs, which is unlikely.

We could either:
- Catch KeyError in structure_wildcat_factory and raise a TypeError in its place: https://github.com/xoeye/typecats/blob/662cf282b734cbbd9a8534ce7f5677225472ba14/typecats/patch.py#L21
- Add could add `KeyError` to `CommonStructuringExceptions, which should be what is actually caught in applications trying to handle structuring errors. Though I'm uncertain whether we are actually attempting to catch TypeError in Stratus or other projects. https://github.com/xoeye/typecats/blob/f200ae9ed4ec2cfd572249a5abb5a1297adeb8de/typecats/types.py#L8

But since this PR already contains breaking changes (GenConverter, AKA this) and is a major version release, we might be able to enable [detailed validation](https://cattrs.readthedocs.io/en/latest/validation.html) which would further enforce a single error group type to be caught going forward, alongside the benefits of detailed validation. This would require a rework of typecats' exception logging

Reply:
If we allow it to be turned off, I think we need to tackle the problem of different exceptions being thrown:

From my last comment:
> We could either:
> 
> * Catch KeyError in structure_wildcat_factory and raise a TypeError in its place: https://github.com/xoeye/typecats/blob/662cf282b734cbbd9a8534ce7f5677225472ba14/typecats/patch.py#L21
> * Add could add `KeyError` to `CommonStructuringExceptions, which should be what is actually caught in applications trying to handle structuring errors. Though I'm uncertain whether we are actually attempting to catch TypeError in Stratus or other projects. https://github.com/xoeye/typecats/blob/f200ae9ed4ec2cfd572249a5abb5a1297adeb8de/typecats/types.py#L8

I'd like to propose a derivation of my above proposed solution by catching the errors in `structure_wildcat_factory` and raising a new `SimpleValidationError` (subclass of `cattrs.BaseValidationError` which would contain the type and message of the original error.